sched/environ/env_dup.c

```diff
@@ -108,7 +108,12 @@ int env_dup(FAR struct task_group_s *group)
             {
               /* Duplicate the parent environment. */
 
-              memcpy(envp, ptcb->group->tg_envp, envlen);
+              if (envcp == NULL)
+                {
+                  envcp = (FAR char * const *)ptcb->group->tg_envp;
+                }
+
+              memcpy(envp, envcp, envlen);
```

Comment:
envp allocated by group_malloc still come from the calling process not the new process. I don't believe this patch can fix the problem totally.

Reply:
It is indeed the new processes addrenv and heap that is in use at this point, please observe `binfmt_execmodule.c` lines 153 onward.

```
#if defined(CONFIG_ARCH_ADDRENV) && defined(CONFIG_BUILD_KERNEL)
/* Instantiate the address environment containing the user heap */

ret = up_addrenv_select(&binp->addrenv, &oldenv);
```

Reply:
Yes, you are right.